src/Chapter2.hs

```diff
@@ -728,7 +762,12 @@ value of the element itself
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate = go
+  where
+    go []     = []
+    go xs = concat $ map (\v -> replicate v v) xs
```

Comment:
Great! 👍🏼 
Note that you don't need a case with `[]` as both `concat` and `map` work with empty lists, so you don't need to handle that! :)

If you want to write this function even more elegantly, you can use the standard function `concatMap` which is a combination of `concat` and `map` as the name suggests 🙂
After that, you could notice, that you would be able to eta-reduce on the last argument :)

Reply:
Updated !